src/main/java/de/unistuttgart/iste/gits/quiz_service/service/QuizService.java

```diff
@@ -131,6 +126,49 @@ public Quiz updateMultipleChoiceQuestion(UUID quizId, UpdateMultipleChoiceQuesti
         });
     }
 
+    /**
+     * Adds a cloze question to a quiz.
+     *
+     * @param quizId the id of the quiz
+     * @param input  the question to add
+     * @return the modified quiz
+     * @throws EntityNotFoundException if the quiz does not exist
+     * @throws ValidationException     if input is invalid, see
+     *                                 {@link QuizValidator#validateCreateClozeQuestionInput(CreateClozeQuestionInput)}
+     */
+    public Quiz addClozeQuestion(UUID quizId, CreateClozeQuestionInput input) {
+        quizValidator.validateCreateClozeQuestionInput(input);
+
+        return modifyQuiz(quizId, entity -> {
+            int number = input.getNumber() == null ? assignNumber(entity) : input.getNumber();
+            checkNumberIsUnique(entity, number);
+
+            QuestionEntity questionEntity = quizMapper.clozeQuestionInputToEntity(input);
+            questionEntity.setNumber(number);
+            entity.getQuestionPool().add(questionEntity);
+        });
+    }
+
+    /**
+     * Updates a cloze question in a quiz.
+     *
+     * @param quizId the id of the quiz
+     * @param input  the updated question
+     * @return the modified quiz
+     * @throws EntityNotFoundException if the quiz or the question does not exist
+     * @throws ValidationException     if input is invalid, see
+     */
+    public Quiz updateClozeQuestion(UUID quizId, UpdateClozeQuestionInput input) {
+        quizValidator.validateUpdateClozeQuestionInput(input);
+
+        return modifyQuiz(quizId, entity -> {
```

Comment:
This isn't the line i want to critique but the other one is from an older PR so I can't review it lol

modifyQuiz calls requireQuizExists even though it already does findById().orElseThrows(), that's unnecessary isn't it? And it makes a second db query which is slow